src/CMakeLists.txt

```diff
@@ -143,6 +143,17 @@ if(APPLE AND (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" OR CMAKE_CXX_COMPILER_
         list(REMOVE_DUPLICATES CMAKE_CXX_IMPLICIT_LINK_LIBRARIES)
 endif()
 
+# Check for base threading library
+set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+set(THREADS_PREFER_PTHREAD_FLAG TRUE)
+find_package(Threads REQUIRED)
+if(${CMAKE_THREAD_LIBS_INIT} MATCHES "-pthread")
```

Comment:
Yeah, I'm not sure on how to improve things: there's a non-documented CMAKE_HAVE_THREADS_LIBRARY variable that would be better than comparing CMAKE_THREAD_LIBS_INIT... but since it's not exposed in doc...